src/test/net_tests.cpp

```diff
@@ -678,7 +678,10 @@ BOOST_AUTO_TEST_CASE(get_local_addr_for_peer_port)
     // Our address:port as seen from the peer, completely different from the above.
     in_addr peer_us_addr;
     peer_us_addr.s_addr = htonl(0x02030405);
-    const CAddress peer_us{CService{peer_us_addr, 20002}, NODE_NETWORK};
+    CAddress peer_us{CService{peer_us_addr, 20002}, NODE_NETWORK};
+    int64_t current_time{GetAdjustedTime()};
+    SetMockTime(current_time);
+    peer_us.nTime = current_time;
```

Comment:
Wouldn't it be better to have the test compare the entire advertised `CAddress`, especially with the difficulties of writing functional test for self-advertisements? 
I know it is not the main intention for this test, but I think it is nice to have this tested on the side, especially with the footgunny `CAddress(CService, ServiceFlags)` ctor that is used so frequently, probably to avoid `int64_t`  / `uint32_t` compiler warnings when inserting a `int64_t` time into the full ctor with `nTime`.
So maybe it would be better to instead also raise the earlier check to the Level of `CAddress`?

Reply:
Alright, it is good to test more, even though this test is targeted at testing the ports stuff. IMO it is better for one test to test one thing, so maybe create a new test that just calls `GetLocalAddrForPeer()` without the ports complications from the `get_local_addr_for_peer_port` test. Anyway, up to you. If the existent test is to be modified, then I guess something like this should suffice:

```diff
     m_node.args->ForceSetArg("-bind", strprintf("3.4.5.6:%u", bind_port));
 
+    const uint32_t current_time = static_cast<uint32_t>(GetAdjustedTime());
+
+    SetMockTime(current_time);
+
     // Our address:port as seen from the peer, completely different from the above.
     in_addr peer_us_addr;
     peer_us_addr.s_addr = htonl(0x02030405);
-    const CAddress peer_us{CService{peer_us_addr, 20002}, NODE_NETWORK};
+    const CAddress peer_us{CService{peer_us_addr, 20002}, NODE_NETWORK, current_time};
 
     // Create a peer with a routable IPv4 address (outbound).
```

Reply:
I took your suggestion to modify, and also changed the check for `expected` to the level of `CAddress`. I think it's natural for a unit test of a function that returns an object to check the entire object is as expected, and not just parts of it, especially if it's just two more fields and a (thorough) separate test that would have caught this issue would have required to replicate some of the port stuff in order to also get to the branch where nTime was left default-initialized.